### ISSUE TITLE
Reviewer 2 comments

### ISSUE THREAD

**Issue by labarba** [barbagroup/jcs_paper_pinn#4]
[Reviewer_comments_for_Vortex_Shedding_PINNs.pdf](https://github.com/barbagroup/jcs_paper_pinn/files/13846042/Reviewer_comments_for_Vortex_Shedding_PINNs.pdf)

**Comment by labarba** [comment]
# Exported to text (mangled equations)




Predictive Limitations of Physics-Informed Neural Networks in Vortex Shedding: Reviewer Comments

This paper addresses an important topic (predictive limitations of PINNs), does a thorough, complete, and transparent analysis as evidence of their claims, and uses good scientific practices and writing. I think this paper is clearly appropriate for the journal, clearly of interest to a broad set of readers, and is close to being accepted for publication. I recommend acceptance with minor revisions. I have two major comments and a handful of minor comments.
Major comments:
•	The abstract states [we] find that data free PINNs are unable to predict vortex shedding.
–	To say that a neural network is “unable” to do something requires a much higher standard of evidence than saying that a neural net- work is “able” to do something. When a neural network achieves some task, it is an existence proof that doing so is possible. When a neural network fails to achieve some task, that doesn’t prove the task is impossible. Perhaps with a different optimizer, or a different architecture, data-free PINNs could predict vortex shedding.
–	There is a second issue, which is that readers excited about the PINN approach are unlikely to agree that this conclusion follows from the experiments, because only the simplest PINN architecture and train- ing strategies were used. As I understand it, much of the research on PINNs is dedicated to discovering tricks that allow PINNs to suc- cessfully solve PDEs that the naive approach cannot solve or doesn’t solve efficiently. To make this conclusion more persuasive, the au- thors should try some of these tricks and show that they don’t work.
–	To some extent, neither of these issues can be completely resolved. Informative and useful null results in machine learning research can and should be accepted for publication, even if they are unable to prove conclusively that positive results are impossible to achieve.
–	To resolve these issues as much as possible, I would like to see an ablation study. Ablation studies are best practice for empirical work in machine learning. See, for example, Winner’s curse? On pace, progress, and empirical rigor by Sculley et al. and Sources of Irre- producibility in Machine Learning: A Review, by Gunderson et al. Learning data-driven discretizations for partial differential equations has a well-done ablation study in the appendix.
–	This ablation study can be added as an additional subsection or appendix; to reduce the amount of required effort, I suggest only performing an ablation study for the data-free, unsteady PINNs and only reporting the lift and drag coefficients (like in either Table 2 or figure 11).



–	While most ablation studies remove components of neural networks to see whether those components are necessary to give a positive result, this ablation study would add components to see if they help give a positive result.
–	At a minimum, the ablation study should try the following to see if they improve performance:
1.	using a so-called ‘prior dictionary’ to enforce the boundary condi- tions, see ModalPINN: An extension of physics-informed Neural Networks with enforced truncated Fourier decomposition for pe- riodic flow reconstruction using a limited number of imperfect sensors.
2.	using different activation functions, including σ(x) = sin x, see Ibid.
3.	using a loss function intended to ‘respect causality’, see Respect- ing causality is all you need for training physics-informed neural networks.
4.	using different hyperparameters.
5.	changing the coefficients of the loss function.
6.	any other tricks that people might use to improve the conver- gence or training of PINNs.
–	I suggest also including the tuning methodology for the chosen hy- perparameters, as this is best practice for empirical work in machine learning (see Winners Curse? by Sculley et al.). Just as it is easy to find irreproducible positive results in machine learning due to badly designed hyperparameter tuning, it is also esay to find irreproducible negative results for the same reason.
•	Reference [7] (Rohrhofer et al.) says that the vortex shedding optimization problem with PINNs “can be resolved, e.g., by using truncated Fourier de- composition with PINNs (Raynaud et al., 2022).” Is it true that Raynaud solves Vortex shedding with data-free PINNs? If so, you should try to re- produce the result, or state prominently that it is possible to solve vortex shedding with data-free PINNs. If not, you should discuss (Raynaud et al., 2022) in the introduction/related work and state how the results from each paper are related.
–	Based on my reading of Raynaud et al., they don’t solve vortex shedding using data-free PINNs. Instead, they use sparse observa- tional data and thus are in a data-driven regime. They also assume that they know the fundamental frequency of vortex shedding, which seems like cheating to me. In any case, you should confirm that my understanding of this paper is correct.
Minor comments:



•	While a numerical approximation (e.g., finite difference) may be a more robust choice . . . I don’t understand why finite difference would be a more ‘robust’ choice than automatic differentiation? If anything, I would expect the opposite to be true. Finite difference (to compute gradients with respect to the inputs of a PINN neural network) would have both rounding error and truncation error, and the step size may not be cho- sen appropriately. Automatic differentiation, by contrast, would return the exact gradient up to numerical precision. Finite difference is likely a more efficient choice than automatic differentiation for computing the derivatives with respect to the input. Though the runtime in both cases would be linear in the size of the neural network, when I compared the two methods the constant factor was higher with AD than with FD. Regard- less of whether AD or FD is used to compute gradients with respect to the inputs of a PINN network, AD will make the procedure for computing the gradients with respect to the parameters of the loss function much easier, and will be dramatically more efficient than FD. In summary: AD is a good tool for PINNs, but the efficiency can be increased at the cost of decreased robustness by using FD to compute the derivatives with respect to the inputs of the PINN.
•	Figure 1 should probably say sum 10
cjL2 for some coefficients cj rather
than sum10 J2, since this is how PINNs normally formulate the loss func-
tion.
•	There are typos in references 27 and 28.
•	reaises is a typo in the abstract.
•	Figures 17 and 18 should label which column is PetIBM and which is the data-driven PINN.
•	The last paragraph of the conclusion is, I believe, based on an incor- rect understanding of the relationship between supervised learning and data-driven PINNs. Data-driven PINNs are meant to be given sparse ob- servations of a physical system known to obey certain physical laws; their goal is to infer the full solution with partial knowledge of the system state and some knowledge of the governing equations. See, for example, Hidden fluid mechanics: Learning velocity and pressure fields from flow visualiza- tions. This capability is unlike anything that supervised learning can do. Thus, the two classes of methods cannot be compared quantitatively in any meaningful way.

**Comment by piyueh** [comment]
Ablation study reference:

- Sculley, D., Snoek, J., Wiltschko, A., & Rahimi, A. (2018). Winner's curse? On pace, progress, and empirical rigor. https://openreview.net/forum?id=rJWF0Fywf
- Gundersen, O. E., Coakley, K., Kirkpatrick, C., & Gil, Y. (2022). Sources of irreproducibility in machine learning: A review. arXiv preprint arXiv:2204.07610. https://arxiv.org/abs/2204.07610
- Bar-Sinai, Y., Hoyer, S., Hickey, J., & Brenner, M. P. (2019). Learning data-driven discretizations for partial differential equations. Proceedings of the National Academy of Sciences, 116(31), 15344-15349. https://www.pnas.org/doi/abs/10.1073/pnas.1814058116

**Comment by labarba** [comment]
Also:

- Wang, S., Sankaran, S. and Perdikaris, P., 2022. Respecting causality is all you need for training physics-informed neural networks. arXiv preprint [arXiv:2203.07404](https://arxiv.org/abs/2203.07404).

**Comment by labarba** [comment]
> The abstract states _[we] find that data free PINNs are unable to predict vortex shedding_.

Yeah, we really need to rephrase that. We really just can say that the PINN method we tried (limited by what the Modulus framework allows) does not predict vortex shedding.

**Comment by piyueh** [comment]
> The abstract states _we find that data free PINNs are unable to predict vortex shedding._
> - To say that a neural network is “unable” to do something requires a much higher standard of evidence than saying that a neural net- work is “able” to do something. When a neural network achieves some task, it is an existence proof that doing so is possible. When a neural network fails to achieve some task, that doesn’t prove the task is impossible. Perhaps with a different optimizer, or a different architecture, data-free PINNs could predict vortex shedding.
> - There is a second issue, which is that readers excited about the PINN approach are unlikely to agree that this conclusion follows from the experiments, because only the simplest PINN architecture and training strategies were used. As I understand it, much of the research on PINNs is dedicated to discovering tricks that allow PINNs to successfully solve PDEs that the naive approach cannot solve or doesn’t solve efficiently. To make this conclusion more persuasive, the authors should try some of these tricks and show that they don’t work.
> - To some extent, neither of these issues can be completely resolved. Informative and useful null results in machine learning research can and should be accepted for publication, even if they are unable to prove conclusively that positive results are impossible to achieve.
> - To resolve these issues as much as possible, I would like to see an ablation study. Ablation studies are best practice for empirical work in machine learning. See, for example, Winner’s curse? On pace, progress, and empirical rigor by Sculley et al. and Sources of Irreproducibility in Machine Learning: A Review, by Gunderson et al. Learning data-driven discretizations for partial differential equations has a well-done ablation study in the appendix.
> - This ablation study can be added as an additional subsection or appendix; to reduce the amount of required effort, I suggest only performing an ablation study for the data-free, unsteady PINNs and only reporting the lift and drag coefficients (like in either Table 2 or figure 11).
> - While most ablation studies remove components of neural networks to see whether those components are necessary to give a positive result, this ablation study would add components to see if they help give a positive result.
> - At a minimum, the ablation study should try the following to see if they improve performance:
>   1. using a so-called ‘prior dictionary’ to enforce the boundary condi- tions, see ModalPINN: An extension of physics-informed Neural Networks with enforced truncated Fourier decomposition for pe- riodic flow reconstruction using a limited number of imperfect sensors.
>   2. using different activation functions, including σ(x) = sin x, see Ibid.
>   3. using a loss function intended to ‘respect causality’, see Respect- ing causality is all you need for training physics-informed neural networks.
>   4. using different hyperparameters.
>   5. changing the coefficients of the loss function.
>   6. any other tricks that people might use to improve the conver- gence or training of PINNs.
> - I suggest also including the tuning methodology for the chosen hyperparameters, as this is best practice for empirical work in machine learning (see Winners Curse? by Sculley et al.). Just as it is easy to find irreproducible positive results in machine learning due to badly designed hyperparameter tuning, it is also esay to find irreproducible negative results for the same reason.

### Reply:

**Comment by piyueh** [comment]
> Reference [7] (Rohrhofer et al.) says that the vortex shedding optimization problem with PINNs “can be resolved, e.g., by using truncated Fourier decomposition with PINNs (Raynaud et al., 2022).” Is it true that Raynaud solves Vortex shedding with data-free PINNs? If so, you should try to re- produce the result, or state prominently that it is possible to solve vortex shedding with data-free PINNs. If not, you should discuss (Raynaud et al., 2022) in the introduction/related work and state how the results from each paper are related.
>- Based on my reading of Raynaud et al., they don’t solve vortex shedding using data-free PINNs. Instead, they use sparse observational data and thus are in a data-driven regime. They also assume that they know the fundamental frequency of vortex shedding, which seems like cheating to me. In any case, you should confirm that my understanding of this paper is correct.

### Reply:

**Comment by piyueh** [comment]
> _While a numerical approximation (e.g., finite difference) may be a more robust choice_ . . . I don’t understand why finite difference would be a more ‘robust’ choice than automatic differentiation? If anything, I would expect the opposite to be true. Finite difference (to compute gradients with respect to the inputs of a PINN neural network) would have both rounding error and truncation error, and the step size may not be cho- sen appropriately. Automatic differentiation, by contrast, would return the exact gradient up to numerical precision. Finite difference is likely a more efficient choice than automatic differentiation for computing the derivatives with respect to the input. Though the runtime in both cases would be linear in the size of the neural network, when I compared the two methods the constant factor was higher with AD than with FD. Regard- less of whether AD or FD is used to compute gradients with respect to the inputs of a PINN network, AD will make the procedure for computing the gradients with respect to the parameters of the loss function much easier, and will be dramatically more efficient than FD. In summary: AD is a good tool for PINNs, but the efficiency can be increased at the cost of decreased robustness by using FD to compute the derivatives with respect to the inputs of the PINN.

### Reply:

**Comment by piyueh** [comment]
> Figure 1 should probably say $\sum\limits_{j=1}^{10} c_j L_j^2$ for some coefficients $c_j$ rather than $\sum\limits_{j=1}^{10} J_j^2$, since this is how PINNs normally formulate the loss function.

### Reply:

**Comment by piyueh** [comment]
> There are typos in references 27 and 28.

### Reply:

**Comment by piyueh** [comment]
> reaises is a typo in the abstract.

### Reply:

**Comment by piyueh** [comment]
> Figures 17 and 18 should label which column is PetIBM and which is the data-driven PINN.

### Reply:

**Comment by piyueh** [comment]
> The last paragraph of the conclusion is, I believe, based on an incorrect understanding of the relationship between supervised learning and data-driven PINNs. Data-driven PINNs are meant to be given sparse observations of a physical system known to obey certain physical laws; their goal is to infer the full solution with partial knowledge of the system state and some knowledge of the governing equations. See, for example, _Hidden fluid mechanics: Learning velocity and pressure fields from flow visualizations_. This capability is unlike anything that supervised learning can do. Thus, the two classes of methods cannot be compared quantitatively in any meaningful way.

### Reply:
